py/client/pydeephaven/agg.py

```diff
@@ -0,0 +1,363 @@
+        cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(avg=_GrpcAggSpec.AggSpecAvg())
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def count_(col: str) -> Aggregation:
+    """Create a Count aggregation. This is not supported in 'Table.agg_all_by'.
+
+    Args:
+        col (str): the column to hold the counts of each distinct group
+
+    Returns:
+        an aggregation
+    """
+    return _AggregationCount(col=col)
+
+
+def partition(col: str, include_by_columns: bool = True) -> Aggregation:
+    """Create a Partition aggregation. This is not supported in 'Table.agg_all_by'.
+
+    Args:
+        col (str): the column to hold the sub tables
+        include_by_columns (bool): whether to include the group by columns in the result, default is True
+
+    Returns:
+        an aggregation
+    """
+    return _AggregationPartition(col=col, include_by_columns=include_by_columns)
+
+
+def count_distinct(cols: Union[str, List[str]] = None) -> Aggregation:
+    """Create a Count Distinct aggregation.
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(count_distinct=_GrpcAggSpec.AggSpecCountDistinct(False))
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def first(cols: Union[str, List[str]] = None) -> Aggregation:
+    """Create a First aggregation.
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(first=_GrpcAggSpec.AggSpecFirst())
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def formula(formula: str, formula_param: str, cols: Union[str, List[str]] = None) -> Aggregation:
+    """Create a user defined formula aggregation.
+
+    Args:
+        formula (str): the user defined formula to apply to each group
+        formula_param (str): the parameter name within the formula
+        cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(formula=_GrpcAggSpec.AggSpecFormula(formula=formula, param_token=formula_param))
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def last(cols: Union[str, List[str]] = None) -> Aggregation:
+    """Create Last aggregation.
```

Comment:
```suggestion
    """Create a Last aggregation.
```
Also present on the server.